### DISCUSSION TITLE
Blend mode issue

### DISCUSSION THREAD

**Discussion by anuragp2018** [google/model-viewer#4914] [Q&A]
when i am trying to use blend mode in particular material it change the original material 
i attached before and after image![Image](https://github.com/user-attachments/assets/4e2c6054-051a-48e9-861b-baffab6b7925)
![Image](https://github.com/user-attachments/assets/def446e7-2311-47f8-8b75-10f835a7004a)

**Comment by elalish** [comment]
Well, changing the blend mode *is* changing the material, so that's working as intended. Blend modes are tricky - probably the worst approximation of PBR in real-time rasterizers. Have you tried turning off double-sided? And checked your triangle winding? Unfortunately there's only so much we can do in a GPU shader.

**Reply by anuragp2018** [reply]
yes i tried turning off double-sided but it not solve my problem

**Reply by anuragp2018** [reply]
Hey I tried turning off double-sided but it not solve my problem any other way to solve this issue actually I want to add model visibility toggle feature in my web app.

**Reply by elalish** [reply]
Can you turn off blending? That seems like the major trouble point.

**Reply by anuragp2018** [reply]
If I turn off blending then how I achieve model visibility feature?

**Reply by elalish** [reply]
Once alpha = 0 it looks okay, right? I would just make the transition short enough that the weird in-between state isn't very visible - maybe fade over 300ms or shorter.
